src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -463,6 +466,77 @@ public Double getPredefinedResult(LfSensitivityFactor lfSensitivityFactor) {
         return connectivityAnalysisResults;
     }
 
+    private void processStation(LfNetwork lfNetwork, HvdcConverterStation<?> station,
+                                Collection<Pair<LfBus, LccConverterStation>> lccs, Collection<Pair<LfBus, LfVscConverterStationImpl>> vscs) {
+        AbstractLfBus bus = (AbstractLfBus) lfNetwork.getBusById(station.getTerminal().getBusView().getBus().getId());
+        if (bus == null) {
+            return;
+        }
+        if (station instanceof LccConverterStation) {
+            lccs.add(Pair.of(bus, (LccConverterStation) station));
+        } else if (station instanceof VscConverterStation) {
+            vscs.add(Pair.of(bus, bus.getGenerators().stream().filter(LfVscConverterStationImpl.class::isInstance)
+                .map(LfVscConverterStationImpl.class::cast)
+                .filter(lfVscConverterStation -> lfVscConverterStation.getId().equals(station.getId()))
+                .findFirst().orElseThrow()));
+        }
+    }
+
+    private Map<LfBus, BusState> applyHvdcContingency(Network network, LfNetwork lfNetwork, PropagatedContingency contingency) {
+        Collection<Pair<LfBus, LccConverterStation>> lccs = new HashSet<>();
+        Collection<Pair<LfBus, LfVscConverterStationImpl>> vscs = new HashSet<>();
+        for (String hvdcId : contingency.getHvdcIdsToOpen()) {
+            HvdcLine hvdcLine = network.getHvdcLine(hvdcId);
+            processStation(lfNetwork, hvdcLine.getConverterStation1(), lccs, vscs);
+            processStation(lfNetwork, hvdcLine.getConverterStation2(), lccs, vscs);
+        }
+        Collection<LfBus> busesToSave = new HashSet<>();
+        busesToSave.addAll(lccs.stream().map(Pair::getKey).collect(Collectors.toSet()));
+        busesToSave.addAll(vscs.stream().map(Pair::getKey).collect(Collectors.toSet()));
+        Map<LfBus, BusState> busStates = BusState.createBusStates(busesToSave);
+
+        for (Pair<LfBus, LccConverterStation> busAndlcc : lccs) {
+            LfBus bus = busAndlcc.getKey();
+            LccConverterStation lcc = busAndlcc.getValue();
+            HvdcLine line = lcc.getHvdcLine();
```

Comment:
This duplicated code is very dangerous, to retrieve the active power set point of a LCC converter station is very dangerous. I am going to try to put a method in the network.